app/services/services/find_for_oauth.rb

```diff
@@ -0,0 +1,32 @@
+module Services
+  class FindForOauth
+    attr_reader :auth
+
+    def initialize(auth)
+      @auth = auth
+    end
+
+    def call
+      authorization = Authorization.where(provider: auth.provider, uid: auth.uid.to_s).first
+      return authorization.user if authorization
+
+      email = auth.info[:email]
+      user = User.where(email: email).first
+      if user
+        user.authorizations.create(provider: auth.provider, uid: auth.uid)
+      elsif email
+        user = create_user_from_oauth(email, auth)
+      end
```

Comment:
```suggestion
      user = User.find_by(email: email)
      if user
        user.authorizations.create(provider: auth.provider, uid: auth.uid)
      else
        user = create_user_from_oauth(email, auth)
      end
```

а вот так нельзя? а то не очень понятные условия, если есть юзер ..., если есть email ...

Reply:
ааа, да так и забыл поправить этот elsif)) Но мне вообще , как то не понравился кака получился этот метод, отрефакторил по другому , разбил на методы приватные . и задействовал create_authorization(auth)